org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/composites/StepModelComposite.java

```diff
@@ -0,0 +1,116 @@
+
+import org.eclipse.richbeans.widgets.scalebox.ScaleBox;
+import org.eclipse.scanning.api.IScannable;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+
+public class StepModelComposite extends Composite {
+
+	private final ScaleBox    start,stop,step,exposureTime;
+
+	public StepModelComposite(Composite parent, int style) {
+		super(parent, style);
+		
+		setLayout(new GridLayout(2, false));
+					
+		Label label = new Label(this, SWT.NONE);
+		label.setText("Start");
+		
+		start = new ScaleBox(this, SWT.NONE);
+		start.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+		start.setUnit("eV");
+		start.setMinimum(0);
+		start.setMaximum(35000);
+		
+		label = new Label(this, SWT.NONE);
+		label.setText("Stop");
+		
+		stop = new ScaleBox(this, SWT.NONE);
+		stop.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+		stop.setUnit("eV");
+		stop.setMinimum(start);
+		stop.setMaximum(35000);
+		
+		label = new Label(this, SWT.NONE);
+		label.setText("Step");
+		
+		step = new ScaleBox(this, SWT.NONE);
+		step.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+		step.setUnit("eV");
+		step.setMinimum(0.00001);
+		step.setMaximum(35000);
+
+		label = new Label(this, SWT.HORIZONTAL|SWT.SEPARATOR);
+		label.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false, 2, 1));
+		
+		label = new Label(this, SWT.NONE);
+		label.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false, 2, 1));
+		label.setText("Exposure overrides the detector configuration");
+		label.setToolTipText("Exposure overrides the detector configuration.\nIt should be set only if you are sure that the exposure of\nthis part of the scan should be different to usual.\n\nNOTE: If set for one region, all regions set an exposure.");
+
+		label = new Label(this, SWT.NONE);
+		label.setText("Exposure");
+		
+		exposureTime = new ScaleBox(this, SWT.NONE);
+		exposureTime.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+		exposureTime.setUnit("s");
+		exposureTime.setMinimum(0);
+		exposureTime.setMaximum(600);
+
+	}
+
+	public ScaleBox getStart() {
+		return start;
+	}
+
+	public ScaleBox getStop() {
+		return stop;
+	}
+
+	public ScaleBox getStep() {
+		return step;
+	}
+
+	public ScaleBox getExposureTime() {
+		return exposureTime;
+	}
+
+	public void setScannable(IScannable<Number> scannable) {
+		setUnit(scannable.getUnit());
+		setBounds(scannable.getMinimum(), scannable.getMaximum());
+	}
+
+	private void setBounds(Number minimum, Number maximum) {
+		start.setMinimum(minimum.doubleValue());
+		start.setMaximum(maximum.doubleValue());
+		stop.setMinimum(start);
+		stop.setMaximum(maximum.doubleValue());		
+		step.setMinimum(0.00001);
+		step.setMaximum(maximum.doubleValue());
+	}
+
+	public void setUnit(String unit) {
+		unit = Optional.of(unit).orElse("");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "unit". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1226)